pkg/querier/queryrange/cache.go

```diff
@@ -0,0 +1,346 @@
+package queryrange
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"sort"
+	"time"
+
+	"github.com/cortexproject/cortex/pkg/chunk/cache"
+	"github.com/cortexproject/cortex/pkg/querier/queryrange"
+	"github.com/cortexproject/cortex/pkg/util/spanlogger"
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/gogo/protobuf/proto"
+	"github.com/gogo/protobuf/types"
+	"github.com/opentracing/opentracing-go"
+	otlog "github.com/opentracing/opentracing-go/log"
+	"github.com/prometheus/common/model"
+	"github.com/uber/jaeger-client-go"
+	"github.com/weaveworks/common/httpgrpc"
+	"github.com/weaveworks/common/user"
+)
+
+type lokiResultsCache struct {
+	logger   log.Logger
+	cfg      queryrange.ResultsCacheConfig
+	next     queryrange.Handler
+	cache    cache.Cache
+	limits   Limits
```

Comment:
This should probably the `queryrange.Limits` from cortex -- it doesn't look like you're using any of the superset methods from the loki `Limits` interface.

Reply:
Limits interface `GenerateCacheKey` is used 
https://github.com/grafana/loki/blob/0c85a4cdf23976114b786dee045eef5ca2b81de6/pkg/querier/queryrange/limits.go#L54